doc/contributing/docs/terms.rst

```diff
@@ -0,0 +1,119 @@
+        to keep it consistent between them.
+
+    term
+        Term is a word, which authors or a particular book, article or documentation set
+        have explicitly selected to denote a :term:`concept` in a particular language
+        and for a particular audience.
+
+        For example, in Tarantool we use the term "[database] sharding" to denote the type of database
+        partitioning mentioned earlier.
+
+Use Preferred Terms
+-------------------
+
+The purpose of using terms is to write concisely and unambiguously,
+which is good for readers.
+But selecting terms is hard.
+Often there are two or more terms for one concept, popular with the community,
+so there's no obvious choice.
+Indeed, selecting and consistently using any of them is much better,
+than not selecting and just using random term each time.
+
+..  glossary::
+
+    restricted term
+        A restricted term is a word which authors have explicitly forbidden to use for denoting a :term:`concept`.
+        Such word is sometimes used as a :term:`term` for the same :term:`concept` elsewhere:
+        in the community, in other books, or in other products' documentation.
+        Sometimes, this word is used to denote a similar, but different concept.
+        In this case the right choice of terms helps us differentiate between concepts.
+
+        For example, in Tarantool we don't use the term "[database] segmentation"
+        to denote what we call "database sharding".
+        Although, other authors could do so.
+        Also, there is a term "[database] partitioning" which we use to denote
+        a wider concept, which is different from sharding.
+
+Define Terms
+------------
+
+Define the term by explaining the concept.
+For key concepts and the ones unique to Tarantool,
+the definition should always be in our documentation.
+
+    Database sharding is a type of horizontal partitioning.
+
+To define a term, use the ``glosary`` directive in the following way:
+
+..  code-block:: rst
+
+    ..  glossary::
+
+        term
+            definition text
+
+        term2
+            definition text
+
+There can be several glossaries in one documentation project.
+A good example of glossary is the one in `Sphinx documentation
+<https://github.com/sphinx-doc/sphinx/blob/master/doc/glossary.rst>`_.
+
+On First Entry, Introduce the Term
+----------------------------------
+
+When you use a term for the first time in a document, introduce it by giving a definition,
+synonyms, translation, examples and links.
+It will help readers learn the term and understand the concept behind it.
+
+
+#.  Define the term or give a link to the definition.
+
+        Database sharding is a type of horizontal partitioning.
+
+    To give a link to the definition, use the `term` role:
+
+    ..  code-block:: rst
+
+        For example, a link to the definition of :term:`concept`.
+        As any role, it can have :term:`custom text <concept>`.
```

Comment:
I would give here an example of the output for the usage of the `term` role like you did for the `abbr` role below.